src/test/java/org/jabref/logic/importer/fetcher/AstrophysicsDataSystemTest.java

```diff
@@ -176,8 +178,7 @@ public void testPerformSearchByIdEmptyDOI() throws Exception {
 
     @Test
     public void testPerformSearchByIdInvalidDoi() throws Exception {
-        Optional<BibEntry> fetchedEntry = fetcher.performSearchById("this.doi.will.fail");
-        assertEquals(Optional.empty(), fetchedEntry);
+        assertThrows(FetcherException.class, () -> fetcher.performSearchById("this.doi.will.fail"));
```

Comment:
I'm not sure about our conventions, but whether we throw an exception or return an empty result should be consistent across id fetchers (and it should be added as a general test for all id fetchers (there should be a test class for all id fetchers))

Reply:
The problem is that our URLDownload throws a FileNotFound Exception on an 404 error, because you are trying to access an input stream of a non existent resource) and thus resulting in a fetcher exception which does not return anything.

Reply:
I think the default is an empty result if the ID cannot be found or used.

Reply:
I changed the method in UrlDownload to return an empty stream when a 404 or 400 error (the latter one is thrown by medline)  is encountered. I simply log the status response code + url .

Reply:
Can't you catch the Exception inside the fetcher? I don't like our URLDownload implementation anyway.

Reply:
Well that was done before but the fetcher exception just encapsulated a file not found exception with no way to getting the underlying http status code.